repos/system_upgrade/el7toel8/actors/repositoriesblacklist/libraries/library.py

```diff
@@ -0,0 +1,38 @@
+from leapp.libraries.stdlib import api
+from leapp.models import  RepositoriesBlacklisted, RepositoriesFacts, RepositoriesMap
+
+
+def get_list_of_optional_repos():
+    """
+        Returns a dictionary with these fields:
+            key = from_id
+            value = to_id
+    """
+    opt_repo = {}
+    repo_map = next(api.consume(RepositoriesMap), None)
+    for repo in repo_map:
+        if repo.from_id.endswith('optional-rpm'):
```

Comment:
There is not repoid with the `optional-rpm` suffix. It's `optional-rpms`.
```suggestion
        if repo.from_id.endswith('optional-rpms'):
```